rust/private/rustc.bzl

```diff
@@ -655,6 +655,18 @@ def establish_cc_info(ctx, crate_info, toolchain, cc_toolchain, feature_configur
             cc_toolchain = cc_toolchain,
             static_library = crate_info.output,
         )
+    elif crate_info.type in ("rlib", "lib"):
+        # bazel hard-codes a check for endswith((".a", ".pic.a",
+        # ".lib")) in create_library_to_link, so we work around that
+        # by creating a symlink to the .rlib with a .a extension.
+        dot_a = ctx.actions.declare_file(crate_info.name + ".a", sibling = crate_info.output)
+        ctx.actions.symlink(output = dot_a, target_file = crate_info.output)
```

Comment:
Is symlink actually safe to use for all platforms? Will windows run into any issues?

Reply:
Not sure about the core rules, but the cargo build script code requires runfiles already: https://docs.bazel.build/versions/master/windows.html#enable-symlink-support

Reply:
Symlink is safe to use:

ctx.actions.symlink() eventually calls https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/windows/WindowsFileSystem.java;l=87
If the target is directory, it creates a junction on Windows, which works just like symlink
If the target is a file, it copies the target if file symlink is not enabled on your machine, otherwise it creates an actual symlink (https://docs.bazel.build/versions/4.0.0/windows.html#enable-symlink-support)

Reply:
AFAICT we're good here.